server/graphql/general/src/queries/log.rs

```diff
@@ -0,0 +1,121 @@
+use async_graphql::*;
+use graphql_core::{
+    generic_filters::EqualFilterStringInput,
+    map_filter,
+    pagination::PaginationInput,
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+use graphql_types::types::{LogConnector, LogNodeType};
+use repository::{EqualFilter, PaginationOption};
+use repository::{LogFilter, LogSort, LogSortField};
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    log::get_logs,
+};
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq)]
+#[graphql(remote = "repository::LogSortField")]
+#[graphql(rename_items = "camelCase")]
+pub enum LogSortFieldInput {
+    Id,
+    LogType,
+    UserId,
+    RecordId,
+}
+
+#[derive(InputObject)]
+pub struct LogSortInput {
+    /// Sort query result by `key`
+    key: LogSortFieldInput,
+    /// Sort query result is sorted descending or ascending (if not provided the default is
+    /// ascending)
+    desc: Option<bool>,
+}
+
+#[derive(InputObject, Clone)]
+pub struct EqualFilterLogTypeInput {
+    pub equal_to: Option<LogNodeType>,
+    pub equal_any: Option<Vec<LogNodeType>>,
+    pub not_equal_to: Option<LogNodeType>,
+}
+
+#[derive(InputObject, Clone)]
+pub struct LogFilterInput {
+    pub id: Option<EqualFilterStringInput>,
+    pub log_type: Option<EqualFilterLogTypeInput>,
+    pub user_id: Option<EqualFilterStringInput>,
+    pub store_id: Option<EqualFilterStringInput>,
+    pub record_id: Option<EqualFilterStringInput>,
+}
+
+#[derive(Union)]
+pub enum LogResponse {
+    Response(LogConnector),
+}
+
+pub fn logs(
+    ctx: &Context<'_>,
+    page: Option<PaginationInput>,
+    filter: Option<LogFilterInput>,
+    sort: Option<Vec<LogSortInput>>,
+) -> Result<LogResponse> {
+    validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            resource: Resource::QueryLog,
```

Comment:
Probably a question for later, but have we thought about this permission model for logs?
Should possibly be related to the store in the log record and other permission mappings?
Maybe should be a new issue for this?

Reply:
yes we briefly discussed it yesterday, with the current log permissions you can see all logs. Could have a version restricted to logs per store though if needed.